Migrate qmsi and esp32 wdt driver api to new generic wdt api interfaces.

SavinayDharmappa opened the pull request.

SavinayDharmappa committed:
drivers: wdt: Migrate qmsi wdt driver api to new wdt api interface.

patch migrates the wdt qmsi driver to new wdt interfaces.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

SavinayDharmappa committed:
test: drivers: watchdog: Modify test application.

test application of watchdog assumes that the variable
created in noinit section will have initial value as
zero. This is an incorrect assumption as variable
in noinit section can have any value. The application
is modified work even if variable in noint section
have any value.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

SavinayDharmappa committed:
drivers: wdt: Migrate esp32 wdt driver api to new wdt interface

patch migrates the esp32 wdt driver to new wdt interfaces

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

nashif commented:
getting this on esp32:

```
***** Booting Zephyr OS v1.13.0-rc3-11-g7e95ac313d *****
Watchdog sample application
Feeding watchdog 5 times
Feeding watchdog...
 ** FATAL EXCEPTION
 ** CPU 0 EXCCAUSE 20 PS 0x00060e20 PC 0x00000000 VADDR 0x00000000
 **  A0 0x80082081  SP 0x3ffb17b0  A2 0x3ffb0000  A3 0x00000004
 **  A4 0x00000000  A5 0x00060c20  A6 0x3ffb0fe8  A7 0x00000028
 **  A8 0x80080811  A9 0x3ffb1760 A10 0x3ffb0000 A11 0x00000000
 ** A12 0x3ffb01f1 A13 0x00000004 A14 0x00000000 A15 0x00000000
 ** LBEG 0x00000000 LEND 0x00000000 LCOUNT 0x00000000
 ** SAR 0x00000018
Current thread ID = 0x3ffb0b00
Faulting instruction address = 0xdeaddead
Fatal fault in ISR! Spinning...
```

nashif commented on drivers/watchdog/wdt_esp32.c:
```diff
@@ -64,44 +65,11 @@ static int wdt_esp32_disable(struct device *dev)
-	 */
+	u32_t ticks = timeout;
 
-	 /* MWDT ticks every 12.5ns.  Set the prescaler to 40000, so the
```
why are you removing those comments?

SavinayDharmappa replied:
ok. I will change it.

SavinayDharmappa commented:
@galak I can do it for quark, But for esp32 still basic dts support needs to be addded. #7581

nashif commented:
@SavinayDharmappa can you please rebase? and please do the DTS support for quark.

SavinayDharmappa committed:
drivers: watchdog: Add dts support for QMSI based watchdog.

patch adds the device binding for qmsi watchdog and device node
for socs which are using wdt qmsi driver.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

SavinayDharmappa commented:
added device tree support for quark
